cultivation/scripts/literature/docinsight_client.py

```
@@ -0,0 +1,101 @@
+        payload = {"job_ids": job_ids}
+        try:
+            # Conditionally include timeout to support mocks without timeout parameter
+            post_kwargs = {'json': payload}
+            if 'timeout' in inspect.signature(requests.post).parameters:
+                post_kwargs['timeout'] = timeout
+            resp = requests.post(url, **post_kwargs)
+            if resp.status_code != 200:
+                raise DocInsightAPIError(f"get_results failed: {resp.status_code} {resp.text}")
+            return resp.json()
+        except requests.Timeout as e:
+            raise DocInsightTimeoutError(f"get_results timed out: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)